modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/StemmerTokenFilterFactory.java

```diff
@@ -165,9 +173,19 @@ public TokenStream create(TokenStream tokenStream) {
 
                 // German stemmers
             } else if ("german".equalsIgnoreCase(language)) {
-                return new SnowballFilter(tokenStream, new GermanStemmer());
+                if (this.version.onOrAfter(IndexVersions.UPGRADE_TO_LUCENE_10_0_0)) {
+                    return new SnowballFilter(tokenStream, new GermanStemmer());
+                } else {
+                    // use pre-L10 GermanStemmer that doesn't normalize umlauts etc...
+                    return new SnowballFilter(tokenStream, new LegacyGermanStemmer());
+                }
             } else if ("german2".equalsIgnoreCase(language)) {
-                // TODO Lucene 10 upgrade: how about bw comp for users relying on german2 stemmer that is now folded into german stemmer?
+                DEPRECATION_LOGGER.warn(
```

Comment:
I am honestly not sure about `warn` vs `critical`, but I have been using `critical` for anything we are planning to remove.

Reply:
I didn't know Critical existed, I was mostly having typical Log4j log levels in mind here. No strong opinion, the description makes me thing "critical" is right here.